tests/v2/validation/nodescaling/scaling_nodepools.go

```diff
@@ -0,0 +1,49 @@
+package nodescaling
+
+import (
+	"testing"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/machinepools"
+	rke1 "github.com/rancher/rancher/tests/framework/extensions/rke1/nodepools"
+	"github.com/rancher/rancher/tests/framework/extensions/workloads/pods"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	ProvisioningSteveResourceType = "provisioning.cattle.io.cluster"
+	defaultNamespace              = "fleet-default"
+)
+
+func ScalingRKE2K3SNodePools(t *testing.T, client *rancher.Client, clusterID string, nodeRoles machinepools.NodeRoles) {
+	cluster, err := client.Steve.SteveType(ProvisioningSteveResourceType).ByID(clusterID)
+	require.NoError(t, err)
+
+	clusterResp, err := machinepools.ScaleMachinePoolNodes(client, cluster, nodeRoles)
+	require.NoError(t, err)
+
+	err = pods.VerifyReadyDaemonsetPods(t, client, cluster)
```

Comment:
I just noticed this sorry! If this func takes a testing.T there is no reason to return an error. Pick one over the other